integration/containerd_image_test.go

```diff
@@ -146,3 +147,38 @@ func TestContainerdImage(t *testing.T) {
 	require.NoError(t, Eventually(checkContainer, 100*time.Millisecond, 10*time.Second))
 	require.NoError(t, Consistently(checkContainer, 100*time.Millisecond, time.Second))
 }
+
+// Test to test the CRI plugin should not see image pulled into containerd in other namespaces.
+func TestContainerdImageInOtherNamespaces(t *testing.T) {
+	testImage := "docker.io/library/busybox:latest"
+	ctx := context.Background()
+
+	t.Logf("make sure the test image doesn't exist in the cri plugin")
+	i, err := imageService.ImageStatus(&runtime.ImageSpec{Image: testImage})
+	require.NoError(t, err)
+	if i != nil {
+		require.NoError(t, imageService.RemoveImage(&runtime.ImageSpec{Image: testImage}))
+	}
+
+	t.Logf("pull the image into containerd")
+	namespacedCtx := namespaces.WithNamespace(ctx, "test")
+	_, err = containerdClient.Pull(namespacedCtx, testImage, containerd.WithPullUnpack)
+	assert.NoError(t, err)
+	defer func() {
+		// Make sure the image is cleaned up in any case.
+		if err := containerdClient.ImageService().Delete(namespacedCtx, testImage); err != nil {
+			assert.True(t, errdefs.IsNotFound(err), err)
+		}
+		assert.NoError(t, imageService.RemoveImage(&runtime.ImageSpec{Image: testImage}))
+	}()
+
+	t.Logf("cri plugin should not see the image")
+	checkImage := func() (bool, error) {
+		img, err := imageService.ImageStatus(&runtime.ImageSpec{Image: testImage})
+		if err != nil {
+			return false, err
+		}
+		return img == nil, nil
+	}
+	require.NoError(t, Consistently(checkImage, 100*time.Millisecond, time.Second))
+}
```

Comment:
should probably add two tests .. 1) make sure we can also pull for our namespace and then see it .. 2) then delete one and make sure it's not visible again but still visible from the ns where it hasn't been deleted.

Reply:
1) That is tested in the other test case `TestContainerdImage`;
2) We can add a case that image deleted in another namespace won't affect cri plugin. Will do.